pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+PEP: 581
+Title: Using GitHub Issues for CPython
+Version: $Revision$
+Last-Modified: $Date$
+Author: Mariatta Wijaya <mariatta@python.org>
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 20-Jun-2018
+
+
+Abstract
+========
+
+This PEP outlines the steps required to migrate Python's issue tracker
+from Roundup to using GitHub Issues.
+
+
+Rationale
+=========
+
+CPython's development moved to GitHub on February 2017. All other projects within
+The PSF's organization are hosted on GitHub and are using GitHub issues.
+CPython is still using Roundup as the issue tracker in bugs.python.org (bpo).
+
+Why GitHub
+----------
+
+It has a lot of nice features readily available out of the box, that are not
+currently available in Roundup / bpo.
+
+- APIs we can use to build integrations and automations.
+
+- Ability to attach screenshots and files in pull requests and issues.
+
+- Administrators and core developers can edit issues comments and pull requests.
+
+- Ability to reply to issues and pull request conversations via email.
+
+- It supports two factor authentication.
+
+- It supports markdown and emoji.
+
+- Core developers don't have to maintain it, giving us more time to focus on
+  the development of Python.
+
+- We can automatically close issues when a PR has been merged [#close-github-issue]_.
+
+- With more than 28 million users, an open source contributor is more likely
+  to already have an account, and familiar with GitHub interface, making it
+  easier to start contributing.
+
+- Email notifications contain metadata [#email-meta]_, integrated with GMail, and
+  allows you to systematically filter emails.
+
+- Email address can remain hidden by choice.
+
```

Comment:
Few more advantages:

* Single namespace for issue numbers and pull request numbers: `#xxx` (instead of `bpo-xxx` vs `GH-xxx`). Many people omits `bpo-` or `GH-` prefix :-(
* Ability to "quote" code using a permalink to lines of code in the Git repository. Example: `https://github.com/python/cpython/blob/bcd3a1a18d841338f57c39f6a7de8cf14d0c3e03/Lib/asyncio/base_events.py#L375-L377` creates:

http://haypo.alwaysdata.net/tmp/quote.png

(The URL to the picture is a temporary link, please copy the picture somewhere!)

Reply:
![quote](https://user-images.githubusercontent.com/3275593/42420415-189667b2-82e3-11e8-8442-01cfe776c8ad.png)

(the image linked to by @vstinner; attached via drag-drop on GitHub's web UI)

Reply:
This image is a png image, not a permalink to the code itself.  Clicking on the image gets you to the image on a screen by itself, not to the code.

Reply:
That was just an example from elsewhere, here's one from this repo: 
https://github.com/python/peps/blob/8ace3c5e1fa67967824fcf2bcf24435ad3cfef2f/pep0/pep.py#L14-L19

Reply:
@terryjreedy The image is a representation of what you get from dropping a link like `https://github.com/python/peps/blob/8ace3c5e1fa67967824fcf2bcf24435ad3cfef2f/genpepindex.py#L17-L27` in a message:

https://github.com/python/peps/blob/8ace3c5e1fa67967824fcf2bcf24435ad3cfef2f/genpepindex.py#L17-L27

Reply:
On bpo, I copy-paste the link and then the quote, between "". 

`https://github.com/python/peps/blob/8ace3c5e1fa67967824fcf2bcf24435ad3cfef2f/genpepindex.py#L4-L5`

"Generating the PEP index is a multi-step process.  To begin, you must first
parse the PEP files themselves, which in and of itself takes a couple of steps:
"
Given that one wants the complete lines, it is definitely easier to only copy-paste the link to get the same effect.

https://github.com/python/peps/blob/8ace3c5e1fa67967824fcf2bcf24435ad3cfef2f/genpepindex.py#L4-L5